packages/core/src/Site/index.ts

```diff
@@ -407,60 +416,55 @@ export class Site {
   }
 
   /**
-   * Adds an about page to site if not present.
+   * Adds default layout files based on the template.
    */
-  async addAboutPage() {
-    const aboutPath = path.join(this.rootPath, ABOUT_MARKDOWN_FILE);
-    try {
-      await fs.access(aboutPath);
-    } catch (error) {
-      if (fs.existsSync(aboutPath)) {
-        return;
-      }
-      await fs.outputFile(aboutPath, ABOUT_MARKDOWN_DEFAULT);
-    }
+  addDefaultLayoutFiles(njkFile: string, njkObject: NunjuckObj | undefined) {
+    const convertedLayoutTemplate = VariableRenderer.compile(
+      fs.readFileSync(path.join(__dirname, njkFile), 'utf8'));
+    const renderedLayout = convertedLayoutTemplate.render(njkObject);
+    const layoutOutputPath = path.join(this.rootPath, LAYOUT_FOLDER_PATH, LAYOUT_DEFAULT_NAME);
+
+    fs.writeFileSync(layoutOutputPath, renderedLayout, 'utf-8');
   }
 
   /**
-   * Adds a footer to default layout of site.
+   * Copies over the contents of substitution files and assigns them to njk variables.
+   *
+   * @param njkSubs The njk variables and their possible file substitutions.
+   * @returns A NunjuckObj with all nunjucks variables in the layout.
    */
-  addDefaultLayoutFiles() {
-    const wikiFooterPath = path.join(this.rootPath, WIKI_FOOTER_PATH);
-    let footer;
-    if (fs.existsSync(wikiFooterPath)) {
-      logger.info(`Copied over the existing ${WIKI_FOOTER_PATH} file to the converted layout`);
-      footer = `\n${fs.readFileSync(wikiFooterPath, 'utf8')}`;
-    }
+  createNjkObjects(templateConfig: TemplateConfig) {
+    const { njkSubs, hasAutoSiteNav, siteNavIgnore } = templateConfig;
+
+    const njkObject: NunjuckObj = {};
+    njkSubs.forEach((njkSub) => {
+      let fileCopy;
+      njkSub.fileSubstitutes.forEach((fileName) => {
+        const subFilePath = path.join(this.rootPath, fileName);
+        if (fs.existsSync(subFilePath)) {
+          logger.info(`Copied over the existing ${fileName} file to the converted layout`);
+          fileCopy = `\n${fs.readFileSync(subFilePath, 'utf8')}`;
+        }
+      });
+      njkObject[njkSub.variableName] = fileCopy;
+    });
 
-    const wikiSiteNavPath = path.join(this.rootPath, WIKI_SITE_NAV_PATH);
-    let siteNav;
-    if (fs.existsSync(wikiSiteNavPath)) {
-      logger.info(`Copied over the existing ${WIKI_SITE_NAV_PATH} file to the converted layout\n`
```

Comment:
I wonder if we can fit in this log somewhere...? I believe this log would still be helpful for the user since `siteNav` is one of our key navigation features.

Reply:
```javascript
  createNjkObjects({ njkSubs, hasAutoSiteNav, siteNavIgnore }: TemplateConfig) {
    const njkObject: NunjuckObj = {};
    njkSubs.forEach((njkSub) => {
      let fileCopy;
      njkSub.fileSubstitutes.forEach((fileName) => {
        const subFilePath = path.join(this.rootPath, fileName);
        if (fs.existsSync(subFilePath)) {
          logger.info(`Copied over the existing ${fileName} file to the converted layout`);
          fileCopy = `\n${fs.readFileSync(subFilePath, 'utf8')}`;
        }
      });
      njkObject[njkSub.variableName] = fileCopy;
    });
```
[lines](https://github.com/WillCWX/markbind/blob/45d0ead81ab4bb4814d04c39fbc570d84632783c/packages/core/src/Site/index.ts#L439-L451)

I have placed this log inside a more general `nunjucks variable` copy function, will that be ok?

Reply:
Ah I should have been clearer, I was also pointing to:

```
+ 'Check https://markbind.org/userGuide/tweakingThePageStructure.html#site-navigation-menus\n'
+ 'for information on site navigation menus.');
```

But looking at the current setup of your functions, it may be more hassle than worth, so the current changes should be fine 👍 

We can re-visit this should we decide that a link to the UG for Site Navigation during conversion is helpful.

Reply:
Ah I should have been clearer, I was also pointing to:

```
+ 'Check https://markbind.org/userGuide/tweakingThePageStructure.html#site-navigation-menus\n'
+ 'for information on site navigation menus.');
```

But looking at the current setup of your functions, it may be more hassle than worth, so the current changes should be fine 👍 

We can re-visit this should we decide that a link to the UG for Site Navigation during conversion is helpful.